### ISSUE TITLE
UIAutomator connection through TCP/IP

### ISSUE THREAD

**Issue by senthilcomm** [openatx/atx-uiautomator#4]
Hi,
I have a requirement that, will need to connect to the tablet(Android) using TCP/IP like 
[adb connect <ipaddress>] using the uiautomator

From the documentation, I can see that we'll need to provide the tablet's serial number. which is as below:
from uiautomator import Device
d = Device('014E05DE0F02000E', adb_server_host='192.168.1.68', adb_server_port=5037)

Could you pls explain, how can I provide the IP address of the tablet in UIAutomator and automate using TCP/IP ?

Thanks,
Senthil

**Comment by codeskyblue** [comment]
Are you mean `adb connect` ?

**Comment by senthilcomm** [comment]
I am using UIAutomator for automating the testcases for the tablet(which runs on Android).
># UIAutomator works fine, if we use serial number of the tablet.
        from uiautomator import Device
        d = Device('014E05DE0F02000E')

**My requirement is as below:**
 I have a scenario where tablet has only IPAddress(no USB connection) and we need to connect the tablet using IPAddress and execute the testcases using UIAutomator
  Could you please let me know if this is possible?, if possible how?

**Comment by codeskyblue** [comment]
it seems you need to learn the command  adb connect

**Comment by senthilcomm** [comment]
Hi shengxiang,

Thanks for the reply.
I think, my explanation of the problem was not clear, let me rephrase below:

Is it possible to use uiautomator over TCP/IP?

**Comment by codeskyblue** [comment]
yes the answer has already told you.

**Comment by senthilcomm** [comment]
Hi Shengxiang,

Thanks for the reply.
From the uiautomator documentation: https://github.com/xiaocong/uiautomator

from uiautomator import Device
d = Device('014E05DE0F02000E', adb_server_host='192.168.1.68', adb_server_port=5037)

I am unable to understand what IP address is for adb_server_host, where should I run adb, in the android tablet?

If you can explain me, what IP is 192.168.1.68 and where is it configured, it can help me to configure appropriately.

Also, I would like to know where to provide the Device(tablet android) IP address in the PC side of UIautomator.

**Comment by codeskyblue** [comment]
First need to ask you a question, how did you make you phone over TCP/IP?

**Comment by senthilcomm** [comment]
I have an Android tablet, which gets ipaddress when connected over wifi or an ethernet cable.

I can see the ipaddress of the tablet in the (Settings-->About --> Status --> IP address)

**Comment by codeskyblue** [comment]
checkout this repo https://github.com/mzlogin/awesome-adb/blob/master/README.md

**Comment by senthilcomm** [comment]
Hi,
Link was useful for adb commands.

Could you pls let me know, how to use Uiautomator with IPaddress of the tablet from PC side with an example.

For your information,  I am using python UIAutomator (https://github.com/xiaocong/uiautomator)

**Comment by codeskyblue** [comment]
中文可以吗？给你发的那个链接里面写的很清楚了都

**Comment by senthilcomm** [comment]
Hi,

I found a fix, made a slight tweak in adb.py and __init__.py and it worked. 

Thanks, nice chatting with you.

**Comment by binel01** [comment]
> Hi,
> 
> I found a fix, made a slight tweak in adb.py and **init**.py and it worked.
> 
> Thanks, nice chatting with you.

Please, I'm facing the same issue you were facing. Please can you tell me how you have done to overcome it ?

**Comment by SumuduAbeysekara** [comment]
Hi senthilcomm,

I'm facing the same issue here. Can you please help?

Thanks
